packages/navigator/src/browser/navigator-model.ts

```diff
@@ -112,4 +162,36 @@ export class FileNavigatorModel extends FileTreeModel {
         return Iterators.cycle(filteredNodes, node);
     }
 
+    protected collectAffectedNodes(change: FileChange, accept: (node: CompositeTreeNode) => void): void {
+        if (this.isFileContentChanged(change)) {
+            return;
+        }
+        this.getNodesByUri(change.uri.parent.toString()).forEach(parentNode => {
+            if (DirNode.is(parentNode) && parentNode.expanded) {
+                accept(parentNode);
+            }
+        });
+    }
+
+    protected isFileContentChanged(change: FileChange): boolean {
+        const roots = this.workspaceService.tryGetWorkspace().roots;
+        if (roots.length === 0) {
+            return false;
+        }
+        const nodeId = WorkspaceRootNode.createId(roots[0].uri, change.uri.toString());
+        const node = this.getNode(nodeId);
+        return change.type === FileChangeType.UPDATED && FileNode.is(node);
+    }
+
+    protected getNodesByUri(nodeUri: string): TreeNode[] {
+        const roots = this.workspaceService.tryGetWorkspace().roots;
```

Comment:
it should be fine to use `.workspace` here, we need to promisify all calles, but it is fine